src/base/kaldi-error.h

```diff
@@ -63,60 +66,51 @@ inline int32 GetVerboseLevel() { return g_kaldi_verbose_level; }
 /// automatically from ParseOptions.
 inline void SetVerboseLevel(int32 i) { g_kaldi_verbose_level = i; }
 
-// Class KaldiLogMessage is invoked from the  KALDI_WARN, KALDI_VLOG and
-// KALDI_LOG macros. It prints the message to stderr.  Note: we avoid
-// using cerr, due to problems with thread safety.  fprintf is guaranteed
-// thread-safe.
-
-// class KaldiWarnMessage is invoked from the KALDI_WARN macro.
-class KaldiWarnMessage {
- public:
-  inline std::ostream &stream() { return ss_; }
-  KaldiWarnMessage(const char *func, const char *file, int32 line);
-  ~KaldiWarnMessage();
- private:
-  std::ostringstream ss_;
+/// Log message severity and source location info.
+struct LogMessageEnvelope {
+  // Positive number cast to LogLevel is for verbose logging.
```

Comment:
sure, squashing makes sense.

On Sun, Apr 17, 2016 at 3:43 PM, Kirill Katsnelson <notifications@github.com
> wrote:

> In src/base/kaldi-error.h
> <https://github.com/kaldi-asr/kaldi/pull/642#discussion_r59994703>:
>
> > -// Class KaldiLogMessage is invoked from the  KALDI_WARN, KALDI_VLOG and
> > -// KALDI_LOG macros. It prints the message to stderr.  Note: we avoid
> > -// using cerr, due to problems with thread safety.  fprintf is guaranteed
> > -// thread-safe.
> > -
> > -// class KaldiWarnMessage is invoked from the KALDI_WARN macro.
> > -class KaldiWarnMessage {
> > - public:
> > -  inline std::ostream &stream() { return ss_; }
> > -  KaldiWarnMessage(const char *func, const char *file, int32 line);
> > -  ~KaldiWarnMessage();
> > - private:
> > -  std::ostringstream ss_;
> > +/// Log message severity and source location info.
> > +struct LogMessageEnvelope {
> > +  // Positive number cast to LogLevel is for verbose logging.
>
> I removed this sloppy comment and updated the following one (LogLevel was
> the née of the enum Severity), plus couple other minor changes: 408c22d
> <https://github.com/kaldi-asr/kaldi/pull/642/commits/408c22d356e21c805be9856f6d4291f9b0eab3de>
>
> Squash?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/642/files/1a5d5b4cf88c8d7c180e4a98372298924eae6973#r59994703>
>